src/veh_interact.cpp

```diff
@@ -316,30 +315,19 @@ task_reason veh_interact::cant_do (char mode)
         } );
         has_tools = true; // checked later
         break;
-    case 'f': // refill mode
-        if (!ptanks.empty()) {
-            std::vector<vehicle_part *>::iterator iter;
-            for (iter = ptanks.begin(); iter != ptanks.end(); ) {
-                if ((*iter)->hp > 0 &&
-                    g->refill_vehicle_part(*veh, *iter, true)) {
-                    pass_checks = true;
-                    iter++;
-                } else {
-                    iter = ptanks.erase(iter);
-                }
-            }
-            if(pass_checks) {
-                return CAN_DO;
-            }
-        }
 
-        // if refillable parts exist but can't be refilled for some reason.
-        if (has_ptank) {
-            return CANT_REFILL;
-        }
+    case 'f': {
+        // first find all possible ammo
+        auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
Good point , I did forget I wrote that although it was with the intention of being certain to never get a nullptr deference. We'd have to check all of the remaining turret code to be sure.

How do you propose to stop such *support* bitrotting? Emphasis mine but in the lack of any actual in-the-wild use examples I think that's a fair assessment.

If we allow non-ammo what should `ammo_current()` and `ammo_type()` return for non-ammo fuels? 

I want to avoid the special case where `ammo_current() == "null"` but `ammo_remaining() > 0` if at all possible. The exists in the `item` code due to legacy items (spray can, lighter etc) but has been the cause of a lot of bugs and adds a lot of boilerplate to the code.